### PR TITLE
Closes #1831 Add more detail to visit and period vignette

### LINKED ISSUE #1831: Documentation:  Add more detail to visit and period vignette
### Please select a category the issue is focused on?

User Guides

### Let us know where something needs a refresh or put your idea here!

From the discussion @bundfussr had on Slack with a user I'm wondering if more could be added to https://pharmaverse.github.io/admiral/articles/visits_periods.html to show the pieces where users need to do their own programming around what admiral offers.

Details Stefan gave in Slack:

At some point study specific code is required to derive period/subperiod variables. There are two options:

- Study specific code is used to derive the variables PxxSwSDT and PxxSwEDT in ADSL. Then create_period_dataset() and derive_vars_joined() can be used to derive period/subperiod variables like ASPER or ASPRSDT in BDS and OCCDS datasets.
- Study specific code is used to derive a dataset with one observations per patient, period, and subperiod (this dataset is called period reference dataset in the vignette). Then derive_vars_period() can be used to derive PxxSwSDT and PxxSwEDT in ADSL and derive_vars_joined() can be used to derive period/subperiod variables like ASPER or ASPRSDT in BDS and OCCDS datasets.

It depends on the specific definition of the periods/subperiods which option works best. If the definition is based on other ADSL variables, I would choose the first option. If the definition is based on vertically structured data like exposure data (EX dataset), I would use the second option.

_I feel worth adding something here as Slack will delete that thread and we might get same question again otherwise._

### PR DESCRIPTION
**Pull request by jeffreyad**
…c Code section.

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by rossfarrugia** [comment]
Thanks for the PR Jeff! I trust Ben and Stefan to do a better job than me reviewing this so will leave it to them.

File: vignettes/visits_periods.Rmd [hunk 1]
```diff
@@ -5,6 +5,9 @@ output:
 vignette: >
   %\VignetteIndexEntry{Visit and Period Variables}
   %\VignetteEngine{knitr::rmarkdown}
+editor_options: 
+  markdown: 
+    wrap: 72
```
Comment (zdz2101) [inline vignettes/visits_periods.Rmd:10]: This is a clever little trick we should encourage everywhere

**Review by zdz2101 — APPROVED** [review]
LGTM!

**Comment by millerg23** [comment]
@jeffreyad looks good to me. I just merged in `devel` branch, if checks run then I will approve.

**Review by zdz2101 — APPROVED** [review]

**Pull request merged**